assignment4.Rmd

```diff
@@ -62,57 +63,81 @@ pca <- prcomp(D2, scale = TRUE)
+plot(D4)
+COR2 <- cor(D4)
+corrplot(COR2, order="AOE", method="circle", tl.pos="lt", type="upper",        
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
 ```
 ## Now print out the eigenvectors (often called loadings) for the components you generated:
 
 ```{r}
-pca$rotation
+pca2 <- prcomp(D4, scale = TRUE)
+pca2$rotation
 
 #Examine the eigenvectors, notice that they are a little difficult to interpret. It is much easier to make sense of them if we make them proportional within each component
 
-loadings <- abs(pca$rotation) #abs() will make all eigenvectors positive
+loadings <- abs(pca2$rotation) #abs() will make all eigenvectors positive
 
 sweep(loadings, 2, colSums(loadings), "/") #sweep() computes each row as a proportion of the column. (There must be a way to do this with dplyr()?)
 
 #Now examine your components and try to come up with substantive descriptions of what some might represent?
 
 #You can generate a biplot to help you, though these can be a bit confusing. They plot the transformed data by the first two components. Therefore, the axes represent the direction of maximum variance. Then mapped onto this point cloud are the original directions of the variables, depicted as red arrows. It is supposed to provide a visualization of which variables "go together". Variables that possibly represent the same underlying construct point in the same direction.  
 
-biplot(pca)
+biplot(pca2)
 
 #Calculate values for each student that represent these your composite variables and then create a new correlogram showing their relationship to mean_correct.
+# The new correlogram shows that prior_percent_correct is more correlated with mean_correct
+(new_corr <- pca2$x %*% t(pca2$rotation))
+corrgram(new_corr)
 
 ```
 # Part III  
 
 ## Also in this repository is a data set collected from class that shows how many students thought that a TC program was related to andother TC program. Use PCA to look for components of related programs. Explain why you think there are relationships between these programs.
+# I decide to choose the first 9  components, because their propotions of variance are larger than 0.03. The larger the proportion of variance, the larger variabilities they can capture. There programs are related because they are either arts or humanities or both. 
+```{r}
+TCprograms <- read.csv("TCprograms.csv",sep=" ")
+pca3 <- prcomp(TCprograms[,-1], scale = TRUE)
+#pca3$rotation
+summary(pca3)
+
+T1 <- TCprograms[,-1]
+T1 <- T1[,1:9]
+COR3 <- cor(T1)
+corrplot(COR3, order="AOE", method="circle", tl.pos="lt", type="upper",        
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+corrgram(T1)
+pca3_2 <- prcomp(T1, scale = TRUE)
+biplot(pca3_2)
```

Comment:
Nice work. It would be good if you provide more interpretation and report for your Part III results.